Add small changes to support new (docker 1.9) networking options

leonidlm opened the pull request.

leonidlm committed:
Set the docker network created by ecs agent to "xplenty"

leonidlm commented:
/cc @motymichaely

motymichaely commented on agent/engine/docker_container_engine.go:
```diff
@@ -317,6 +317,10 @@ func (dg *DockerGoClient) createContainer(ctx context.Context, config *docker.Co
 
+	// Hardcode the network we are going to use (xplenty/overlay)
+	// For this agent to work we must create this network (once) first.
+	hostConfig.NetworkMode = "xplenty"
```
@leonidlm - Just wondering - couldn't we pass this to `docker create` call as part of it's host configuration file?

leonidlm committed:
Make the container hostname to be identical to it's name

We need this to allow tasktrackers to be able to reach each other by
their hostname.
